dev-infra/commit-message/restore-commit-message.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {info} from 'console';
+import {writeFileSync} from 'fs';
+import {resolve} from 'path';
+
+import {getRepoBaseDir} from '../utils/config';
+
+import {loadCommitMessageDraft} from './commit-message-draft';
+
+/**
+ * Restore the commit message draft to the git to be used as the default commit message.
+ */
+export function restoreCommitMessage(filePath: string, source: string) {
+  if (!!source) {
+    info('Skipping commit message restoration attempt');
+    if (source === 'message') {
+      info('A commit message was already provided via the command with a -m or -F flag');
+    }
+    if (source === 'template') {
+      info('A commit message was already provided via the -t flag or config.template setting');
+    }
+    if (source === 'squash') {
+      info('A commit message was already provided as a merge action or via .git/MERGE_MSG');
+    }
+    if (source === 'commit') {
+      info('A commit message was already provided via another commit via --fixup, -c, -C');
```

Comment:
Nit suggestion (if it makes sense): Might sound little better if we avoid the double `via`:

```suggestion
      info('A commit message was already provided through a revision specified via --fixup, -c, -C');
```